src/Test/Proxy/FastlyProxy.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\Test\Proxy;
+
+class FastlyProxy extends AbstractProxy
```

Comment:
is this WIP?

Reply:
Yes, I'm unsure what we want to do on testing.

@vvuksan sent me tips on maybe using this: https://fiddle.fastlydemo.net/
But I'm unsure if it is suitable for automated functional testing.
So I'm wondering if it's best to just aim for unit testing for now.

@vvuksan Is that what you had in mind as well? Or do you have suggestions on how we can also use fiddle for automated functional tests (CI/Travis)?

Reply:
I have the same issue for my LiteSpeed PR (which I should polish, I know...). OpenLiteSpeed is no issue because you can install it during CI but the enterprise version (which happens to not be the same code base somehow) cannot be installed because one needs a license. I've already talked to them asking for some kind of OSS license but no luck so far. I guess we'll have certain integrations without functional tests then :(

Reply:
If you need ESI, then [indeed it seems](https://www.litespeedtech.com/products/litespeed-web-server/editions) you won't be able to run the tests on the open source version.

Reply:
We don't need ESI for our tests but the thing is, you just cannot install the Enterprise version due to the license. And there's no guarantee things work because they do in the OSS variant as they share no code. It might sound weird (at least to me it does) but the implementations are in fact completely different :) Anyway, not the discussion I wanted to start here. I just wanted to show another example of where functional tests on CI are probably not going to happen.

Reply:
yes, my point was that the code difference is not our issue, it's theirs. So if they behave different that is a bug on their side (imho). Aka, unless you rely on enterprise features we can "safely" use oss version for functional.

Reply:
Ah, I agree yes :)